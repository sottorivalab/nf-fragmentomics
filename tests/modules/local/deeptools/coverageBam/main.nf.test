nextflow_process {

    name "Test Process COVERAGEBAM"
    script "modules/local/deeptools/coverageBam/main.nf"
    process "COVERAGEBAM"

    test("Should run COVERAGEBAM without failures") {

        when {
            params {
                outdir = "tests/results"
            }

            process {
                """
                input[0] = [
                    [ caseid:'test', sampleid: 'test', timepoint: 'T1' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.blacklist_intervals.bed', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.bw, process.out.versions).match()
        }

    }

}
