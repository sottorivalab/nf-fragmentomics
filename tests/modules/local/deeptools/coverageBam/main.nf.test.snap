{
    "Should run COVERAGEBAM without failures": {
        "content": [
            [
                [
                    {
                        "caseid": "test",
                        "sampleid": "test",
                        "timepoint": "T1"
                    },
                    "test.paired_end.sorted.bw:md5,abf9856dc3f59565bbf55f89194d54d8"
                ]
            ],
            [
                "versions.yml:md5,47275d818815391772f1e942b5ec305a"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-23T10:33:19.760774"
    }
}